common/softwarecontainer-common.cpp

```diff
@@ -47,6 +47,23 @@ bool isDirectory(const std::string &path)
     return S_ISDIR(getStat(path).st_mode);
 }
 
+bool isDirectoryEmpty(const std::string &path) {
+  int n = 0;
+  struct dirent *d;
+  DIR *dir = opendir(path.c_str());
+  if (dir == NULL) //Not a directory or doesn't exist
+    return true;
+  while ((d = readdir(dir)) != NULL) {
+    if(++n > 2)
```

Comment:
There probably are, but this is fairly simple at least.. I could add a comment about it at least?